octopipes/benchmark.py

```diff
@@ -9,9 +9,15 @@
 
 class Benchmark:
     def __init__(self, dataloader: Dataloader, workflows: list[Workflow],
-                 flows_factory: AggregateFlowsFactory | None = None) -> None:
+                 flows_factory: AggregateFlowsFactory | None = None, mode: str = "threaded") -> None:
+
+        """
+        :param str mode: "single" for sequential execution, "threaded" for multithreading.
```

Comment:
It might a better design choice to select the "run_time" mode by checking the actual batch size of the dataloader. In the case the batch is 1, sequential, threaded otherwise.